packages/webapp/src/pages/play.tsx

```diff
@@ -105,14 +106,27 @@ const Play: FablePage = ({ isHydrated }) => {
   const cancellationHandler = useCancellationHandler(loading)
 
   const cantDrawCard = cantTakeActions || gameData.currentStep !== GameStep.DRAW
-  const doDrawCard = useCallback(
-    () => drawCard({
-        gameID: gameID!,
-        playerAddress: playerAddress!,
-        setLoading,
-        cancellationHandler
-      }),
-    [gameID, playerAddress, setLoading, cancellationHandler])
+  const doDrawCard = useCallback(() =>
+    drawCard({
+      gameID: gameID!,
+      playerAddress: playerAddress!,
+      setLoading,
+      cancellationHandler,
+    }),
+  [gameID, playerAddress, setLoading, cancellationHandler])
+
+  useEffect(() => {
+    // Automatically submit the card draw transaction when it's our turn
+    if (gameData && currentPlayer(gameData) === playerAddress && !cantDrawCard) {
+      toast.promise(doDrawCard, {
+        loading: "Your Turn - Drawing Card...",
+        success: () => {
+          return "Card Drawn Successfully!"
+        },
+        error: "Error Drawing Card",
```

Comment:
We need to handle this better. The `drawCard` function will already call `setError` (in `write.ts`) to trigger the display of a global modal displaying the error. I think it's probably good to keep this for now. Can we just hide the toast in case of error?

Also if an error occurs, there's a question of what to do about card draw. The current implem will be stuck because the dependencies of the `useEffect` are not expected to change. But if the error is due to a temporary RPC outage, this is a problem.

We'll need to redisplay the draw button in this case. I'd do this by checking the return value of `drawCard` inside `doDrawCard` and if negative, set a state variable that controls the display of the old draw button.